neres/cli.py

```diff
@@ -1,15 +1,313 @@
+import sys
+import os
+import platform
+import subprocess
+
 
 import click
+import humanize
+from terminaltables import SingleTable
+
+import newrelic
+import urls
+from spinner import Spinner
+
+
+@click.command()
+@click.argument('name')
+@click.argument('uri')
+@click.option('--location', default=['AWS_US_WEST_1'], multiple=True)
+@click.option('--frequency', default=10)
+@click.option('--email', default=[''], multiple=True)
+@click.option('--validation-string', default='')
+@click.option('--bypass-head-request', default=False, is_flag=True)
+@click.option('--verify-ssl', default=False, is_flag=True)
+@click.option('--redirect-is-failure', default=False, is_flag=True)
+@click.option('--sla-threshold', default=7)
+@click.option('--raw', default=False, is_flag=True)
+def add_monitor(name, uri, location, frequency, email, validation_string,
+                bypass_head_request, verify_ssl, redirect_is_failure, sla_threshold, raw):
+    if validation_string:
+        # We must bypass head request we're to validate string.
+        bypass_head_request = True
+
+    with Spinner('Creating monitor: ', remove_message=raw):
+        status, message, monitor = newrelic.create_monitor(name, uri, frequency, location, email,
+                                                           validation_string, bypass_head_request,
+                                                           verify_ssl, redirect_is_failure,
+                                                           sla_threshold)
+
+    if raw:
+        print(monitor)
+        return
+
+    if status == 0:
+        print(click.style(u'OK', fg='green', bold=True))
+        print('Monitor: ' + message)
+    else:
+        print(click.style(u'Error', fg='red', bold=True))
+        raise click.ClickException(message)
+
+
+@click.command()
+@click.argument('monitor')
+@click.option('--name', default=None)
+@click.option('--uri', default=None)
+@click.option('--add-location', 'add_locations', default=None, multiple=True)
+@click.option('--clear-locations', default=False, is_flag=True)
+@click.option('--remove-location', 'remove_locations', default=None, multiple=True)
+@click.option('--add-email', 'add_emails', default=None, multiple=True)
+@click.option('--remove-email', 'remove_emails', default=None, multiple=True)
+@click.option('--clear-emails', default=False, is_flag=True)
+@click.option('--frequency', default=None)
+@click.option('--sla-threshold', 'slaThreshold', default=None)
+@click.option('--validation-string', default=None)
+@click.option('--no-validation-string', default=None, is_flag=True)
+@click.option('--bypass-head-request/--no-bypass-head-request', default=None, is_flag=True)
+@click.option('--verify-ssl/--no-verify-ssl', default=None, is_flag=True)
+@click.option('--redirect-is-failure/--no-redirect-is-failure', default=None, is_flag=True)
+@click.option('--raw', default=False, is_flag=True)
+def update_monitor(monitor, **kwargs):
+    if kwargs['no_validation_string']:
```

Comment:
`click` will always populate the variables with default or user provided values, so the key will by always there.